profiles/views.py

```diff
@@ -52,40 +52,32 @@ class SavedCompaniesCreate(CreateAPIView):
     serializer_class = SavedCompanySerializer
     pagination_class = ForumPagination
 
-    def post(self, request):
+    def post(self, request, *args, **kwargs):
         user = request.user
-        pk = request.data.get("company_pk")
-
-        # Check if the company is already in the user's saved list
-        if SavedCompany.objects.filter(user=user, company_id=pk).exists():
-            saved_company_destroyer = SavedCompaniesDestroy()
-            return saved_company_destroyer.destroy(request, pk)
-
-        serializer = SavedCompanySerializer(
-            data={"company": pk, "user": user.id}
-        )
-        if serializer.is_valid():
-            serializer.save()
-            return Response({"Company added": serializer.data})
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        pk = request.data.get("company")
+
+        saved_company = SavedCompany.objects.filter(
+            user=user, company_id=pk
+        ).first()
+        if saved_company:
+            return redirect(
+                "profiles:saved_companies_destroy", pk=saved_company.pk
+            )
+        request.data.update({"user": user.id})
+        return self.create(request, *args, **kwargs)
 
 
 class SavedCompaniesDestroy(DestroyAPIView):
     """
     Remove the company from the saved list.
     """
 
+    allowed_methods = ("GET", "DELETE")
+    queryset = SavedCompany.objects.all()
     permission_classes = [IsAuthenticated]
 
-    def destroy(self, request, pk):
-        user = request.user
-        saved_company = get_object_or_404(
-            SavedCompany, company_id=pk, user=user
-        )
-        saved_company.delete()
-        return Response(
-            f"Company {pk} deleted", status=status.HTTP_204_NO_CONTENT
-        )
+    def get(self, request, *args, **kwargs):
+        return super().destroy(request, *args, **kwargs)
```

Comment:
Please explain, what is going on here.

Reply:
We added get() method because we redirecting to this view and redirected request has GET method by default.   This method does exactly the same what delete() do: evokes destroy()